### ISSUE TITLE
Memory question

### ISSUE THREAD

**Issue by ZhaofengWu** [ShannonAI/CorefQA#10]
I'm trying to re-implement your architecture, but got some OOM issues. Since you're passing the question+context through the transformer for every single potential mention, this effectively multiplies the batch size by the number of potential mentions (and even double on top of that if dynamic half-length windows are used), which can be quite large. How did you handle this issue? The only thing I can think of is to not require gradient for the transformer in the QA module.

**Comment by pku-wuwei** [comment]
Yes, you are right, the memory requirement is quite large. We actually implemented it on a Google Cloud TPU pod.

**Comment by yuvalvarkel** [comment]
@pku-wuwei what TPU configuration did you use? Did you use the same code version as published here?

I tried running it on a TPU and it fails since the file_based_input_fn_builder returns features with dynamic shape, which are not supported on TPU. Am I missing something?
